src/aws/ec2-wrapper.ts

```diff
@@ -105,19 +89,22 @@ export class EC2Wrapper {
   /**
    *  Returns the promise that contains the security group IDs
    * @param {string} vpcId
-   * @param {string[]} securityGroupNames
+   * @param {string[]} names
+   * @param {string[]} tagKey
+   * @param {string[]} tagValues
    * @returns {Promise.<string[]>}
    */
-  public async getSecurityGroupIds (vpcId: string, securityGroupNames: string[]): Promise<string[]> {
-    const params = {
-      Filters: [{
-        Name: "vpc-id",
-        Values: [vpcId]
-      }, {
-        Name: "group-name",
-        Values: securityGroupNames
-      }]
-    };
+  public async getSecurityGroupIds (vpcId: string, names?: string[], tagKey?: string, tagValues?: string[]): Promise<string[]> {
```

Comment:
Maybe this should be two separate functions like `getSecurityGroupIdsByTag` and `getSecurityGroupIdsByName`

Reply:
I will update it with https://www.serverless.com/framework/docs/providers/aws/guide/plugins/#extending-validation-schema in the next release